### PR TITLE
feat(dds): move dds resources to services path

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

move dds resources to services path

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes NONE

**Special notes for your reviewer**:

**Release note**:

```release-note
move dds resources to services path
```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dds' TESTARGS='-run TestAccDDSFlavorV3DataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dds -v -run TestAccDDSFlavorV3DataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccDDSFlavorV3DataSource_basic
=== PAUSE TestAccDDSFlavorV3DataSource_basic
=== CONT  TestAccDDSFlavorV3DataSource_basic
--- PASS: TestAccDDSFlavorV3DataSource_basic (56.60s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dds       56.690s

make testacc TEST='./huaweicloud/services/acceptance/dds' TESTARGS='-run TestAccDDSV3Instance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dds -v -run TestAccDDSV3Instance_basic -timeout 360m -parallel 4
=== RUN   TestAccDDSV3Instance_basic
=== PAUSE TestAccDDSV3Instance_basic
=== CONT  TestAccDDSV3Instance_basic
--- PASS: TestAccDDSV3Instance_basic (2911.50s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dds       2911.573s
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -14,5 +14,6 @@ require (
 	github.com/smartystreets/goconvey v0.0.0-20190222223459-a17d461953aa // indirect
 	github.com/stretchr/testify v1.7.0
 	github.com/unknwon/com v1.0.1
+	golang.org/x/net v0.0.0-20210326060303-6b1517762897
```
Comment (ShiChangkuo) [inline go.mod]: why we need this package?

File: huaweicloud/services/acceptance/acceptance.go [hunk 2]
```diff
@@ -249,7 +250,8 @@ func (rc *resourceCheck) CheckResourceExists() resource.TestCheckFunc {
 					rc.resourceName, rs.Primary.ID, err)
 			}
 			if rc.resourceObject != nil {
-				rc.resourceObject = r
+				b, _ := json.Marshal(r)
+				json.Unmarshal(b, rc.resourceObject)
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/acceptance.go]: suggest to push in another PR

File: huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go [hunk 3]
```diff
@@ -1,32 +1,66 @@
...
+	insts := instances.Instances
+	found := insts[0]
+	return &found, nil
+
+}
+
 func TestAccDDSV3Instance_basic(t *testing.T) {
 	var instance instances.InstanceResponse
-	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
+	rName := acceptance.RandomAccResourceName()
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go:45]: `WithDash()`

File: huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go [hunk 4]
```diff
@@ -82,100 +116,38 @@ func TestAccDDSV3Instance_basic(t *testing.T) {
 
 func TestAccDDSV3Instance_withEpsId(t *testing.T) {
 	var instance instances.InstanceResponse
-	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
+	rName := acceptance.RandomAccResourceName()
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go:118]: ditto

File: huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go [hunk 5]
```diff
@@ -82,100 +116,38 @@ func TestAccDDSV3Instance_basic(t *testing.T) {
...
-		return nil
-	}
-}
-
 func testAccCheckDDSV3InstanceFlavor(instance *instances.InstanceResponse, groupType, key string, v interface{}) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
+		log.Printf("xxxxx: %v", instance)
+		log.Printf("xxxxx: %v", &instance)
+		log.Printf("xxxxx: %v", *instance)
+
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go]: Is it debug prints for coding?

Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go]: fixed

File: go.mod (hunk 1 above)
Comment (Jason-Zhang9309) [reply go.mod]: fixed

File: huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go (hunk 3 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go:45]: don't need, RandomAccResourceName() is fine

File: huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go (hunk 4 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go:118]: don't need, RandomAccResourceName() is fine

File: huaweicloud/services/acceptance/acceptance.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/acceptance.go]: another PR #1685 is created

File: huaweicloud/services/acceptance/dds/data_source_huaweicloud_dds_flavors_v3_test.go [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+				),
+			},
+		},
+	})
+}
+
+var testAccDDSFlavorV3DataSource_basic = `
+data "huaweicloud_dds_flavors" "flavor" {
+    engine_name = "DDS-Community"
+    vcpus = 8
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/dds/data_source_huaweicloud_dds_flavors_v3_test.go]: When multiple arguments with single-line values appear on consecutive lines at the same nesting level, align their equals signs.

File: huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go [hunk 7]
```diff
@@ -1,32 +1,65 @@
...
+	instances, err := instances.ExtractInstances(allPages)
+	if err != nil {
+		return nil, err
+	}
+	if instances.TotalCount == 0 {
+		return nil, fmtp.Errorf("dds instance not found.")
+	}
+
+	insts := instances.Instances
+	found := insts[0]
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go:38]: suggest to extract this part of the code as a function, the same code has been declared in `resource_huaweicloud_dds_instance_v3.go`.

File: huaweicloud/services/acceptance/dds/data_source_huaweicloud_dds_flavors_v3_test.go (hunk 6 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/dds/data_source_huaweicloud_dds_flavors_v3_test.go]: fixed

File: huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go (hunk 7 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go:38]: Although the code looks similar, the returned content is different. If we extract this as a function, only line25~32 can be used in the func. I don't think it is necessary.

File: huaweicloud/services/acceptance/dds/data_source_huaweicloud_dds_flavors_v3_test.go [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDDSFlavorV3DataSource_basic,
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(
+						dataSourceName, "engine_name", "DDS-Community"),
```
Comment (chengxiangdong) [inline huaweicloud/services/acceptance/dds/data_source_huaweicloud_dds_flavors_v3_test.go]: this is not a valid check, it is recommended to add a check for `flavors`.

Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/dds/data_source_huaweicloud_dds_flavors_v3_test.go]: fixed

**Pull request merged**